src/Shared/ServerCertificateValidationHandler.cs

```diff
@@ -5,33 +5,35 @@
 
 using System;
 using System.Net.Http;
+using OpenTelemetry.Internal;
 
-namespace OpenTelemetry.ResourceDetectors.AWS.Http;
+namespace OpenTelemetry.ResourceDetectors;
 
-internal class Handler
+internal class ServerCertificateValidationHandler
 {
-    public static HttpClientHandler? Create(string certificateFile)
+    public static HttpClientHandler? Create(string certificateFile, IServerCertificateValidationEventSource? log = null)
```

Comment:
Logging wasn't really optional before; I'm thinking `log` should be a required parameter.  

I don't want future maintainers to think that `log` is optional, unless there is a clear (and documented) usecase as to when it doesn't need to be provided.